examples/components/CCFRAUD_VERTICAL/traininsilo/datasets.py

```diff
@@ -0,0 +1,124 @@
+import torch
+from torch.utils.data import Dataset
+
+
+class FraudDataset(Dataset):
+    """FraudDataset Dataset - combination of features and labels
+
+    Args:
+        feature: Transaction detail tensors
+        target: Tensor of labels corresponding to features
+
+    Returns:
+        None
+    """
+
+    def __init__(self, df):
+        if "is_fraud" in df.columns:
```

Comment:
here as well, I'd rather clarify the design by, instead of a test like this, having two distinct FraudDataset classes, one for the host and one for the contributor.

Overall, I propose we decouple much more what's happening on the host side from what's happening in the contributor side. Ideally, they would share only the communication layer?

Reply:
Looking at models.py and in there you have distinct models, of course. I'd suggest having the same approach for the datasets.

Reply:
@jfomhover for models it makes sense to decouple them as they are completely independent and they do not share anything. However, for the dataset, the difference is only in x/y availability. Even when you are writing dataset for very simple case you want to have it versatile to yield only sample or only labels or both depending on availability.